cluster.go

```diff
@@ -51,7 +51,7 @@ type member struct {
 }
 
 // newCluster returns a new marathon cluster
-func newCluster(client *http.Client, marathonURL string) (*cluster, error) {
+func newCluster(client *http.Client, marathonURL string, DCOSPath string) (*cluster, error) {
```

Comment:
This looks out of place to me: We should not need to add a DCOS-specific parameter to the function responsible for creating the cluster configuration. Besides the design worries, it also adds a lot of noise to multiple call sites now and in the future.

What I think would be better (but not ideal; see below) is to keep adding the DCOS path in `apiCall()`. With the rest of your PR, it should boil down to something like

```
if r.config.DCOSToken != "" {		
   url = fmt.Sprintf("%s/%s/%s", member, r.config.DCOSPath, uri)
}
```

(This should work because we made sure `r.config.DCOSPath` is always set.)

Taking a step back and thinking about how we implemented the default `/marathon` path for DCOS in the first place, however, I am wondering if that was the best approach back then. In particular, does anything speak against requiring users to specify the path on the Marathon URL string? That is, shouldn't it work just as well if users passed `http://marathon1:8080/marathon,marathon2:8080/marathon,marathon3:8080/marathon` (or any path other than `marathon`) to `Config.URL`? I think this would remove a lot of extra code/boilerplate and save us from extending the `Config` struct at the price of just a little redundancy at the user's end.
The only part making things slightly more complicated is that we already default to `/marathon` if a DCOS token is given. I think we can continue to handle this properly though by checking during `apiCall()` if a DCOS token is given **and** a path is missing; in this case, we'd continue to add `/marathon` (and otherwise expect the user to have provided a custom DCOS path).

@mariomarin what do you think?